### PR TITLE
Implement static loading skeleton

### LINKED ISSUE #16473: Refactor loading skeleton
### Summary

Refactor loading skeleton into view and style file and also add support for static loading skeleton instead of animated one.
More details: https://github.com/status-im/status-mobile/pull/16438#issuecomment-1619018513

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes: https://github.com/status-im/status-mobile/issues/16473

### Summary

PR implements a loading skeleton as per Figma design parameters and also adds missing content types.
PR just adds a static loading skeleton, animation of the loading skeleton will be implemented separately.
Once the animation is implemented we can remove the current implementation, until then we have to keep both.
(will create an issue for that)

### Testing
Manual testing doesn't require, PR just implements the quo2 component.

status: ready

### CODE REVIEW

File: src/status_im2/contexts/quo_preview/main.cljs [hunk 1]
```diff
@@ -67,7 +67,6 @@
     [status-im2.contexts.quo-preview.notifications.toast :as toast]
     [status-im2.contexts.quo-preview.onboarding.small-option-card :as small-option-card]
     [status-im2.contexts.quo-preview.password.tips :as tips]
-    [status-im2.contexts.quo-preview.posts-and-attachments.messages-skeleton :as messages-skeleton]
```
Comment (Parveshdhull) [inline src/status_im2/contexts/quo_preview/main.cljs:70]: Probably outdated, currently just a blank screen. Animated skeleton will be part of new preview screen

File: src/quo2/components/loaders/skeleton/view.cljs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                              blur?           colors/white-opa-5
+                              (= theme :dark) colors/neutral-90
+                              :else           colors/neutral-5)]
+    [rn/view {:style {:padding 8}}
+     (doall
+      (for [index (range number-of-skeletons)]
+        ^{:key index}
+        [skeleton-item (mod index 4) content color]))]))
+
+(def view (theme/with-theme internal-view))
```
Comment (J-Son89) [inline src/quo2/components/loaders/skeleton/view.cljs:46]: 🙌

File: src/quo2/components/loaders/skeleton/style.cljs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+   :width            32
+   :border-radius    16
+   :background-color color})
+
+(def content-container
+  {:padding-left 8})
+
+(defn content-view
+  [{:keys [type index content color]}]
+  (let [{:keys [width height margin-bottom margin-top]}
```
Comment (smohamedjavid) [inline src/quo2/components/loaders/skeleton/style.cljs]: `margin-top` is not in `constants/content-dimensions`.

File: src/quo2/core.cljs [hunk 4]
```diff
@@ -113,6 +114,7 @@
 (def checkbox quo2.components.selectors.selectors.view/checkbox)
 (def filter quo2.components.selectors.filter.view/view)
 (def skeleton quo2.components.loaders.skeleton/skeleton)
+(def static-skeleton quo2.components.loaders.skeleton.view/view)
```
Comment (J-Son89) [inline src/quo2/core.cljs]: could you make a loaders section and move it to that? same for other `loaders` in this file!

File: src/quo2/components/loaders/skeleton/view.cljs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+               :content content
+               :color   color})}]
+    (when (= content :notifications)
+      [rn/view
+       {:style (style/content-view {:type    :message2
+                                    :index   index
+                                    :content content
+                                    :color   color})}])]])
+
+(defn internal-view
```
Comment (smohamedjavid) [inline src/quo2/components/loaders/skeleton/view.cljs]: Small Improvement: Updating `internal-view` as private would prevent using it without theme context.

File: src/status_im2/contexts/quo_preview/loaders/skeleton.cljs [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    :options [{:key   :list-items
+               :value "List items"}
+              {:key   :notifications
+               :value "Notifications"}
+              {:key   :messages
+               :value "Messages"}]}
+   {:label "Blur?"
+    :key   :blur?
+    :type  :boolean}
+   {:label "Animated?" ;; Not implemented yet
```
Comment (smohamedjavid) [inline src/status_im2/contexts/quo_preview/loaders/skeleton.cljs]: Is it fine to not have this option until the animation is implemented?

**Review by smohamedjavid — COMMENTED** [review]
@Parveshdhull - Thank you for the PR!

I found that the skeleton works only on `Light mode` and `Dark mode + blur`.  Is this expected? I believe this is not the case. Correct me if i'm wrong.

<table>
  <tr>
    <th>
    Light Mode
    </th>
    <th>
      Light Mode + Blur
    </th>
<th>
     Dark Mode
    </th>
<th>
      Dark Mode + Blur
    </th>
  </tr>
  <tr>
    <td>
      
    </td>
    <td>
      
    </td>
    <td>
      
    </td>
   <td>
      
    </td>
  </tr>
</table>

**Review by J-Son89 — APPROVED** [review]
Looks great, nice work @Parveshdhull! 
Consider adding a few component spec test if possible too. 👍

**Review by smohamedjavid — APPROVED** [review]

**Comment by Parveshdhull** [comment]
Hi @smohamedjavid and @J-Son89, Thank you very much for reviewing PR.
Created a follow-up issue https://github.com/status-im/status-mobile/issues/16482

> I found that the skeleton works only on Light mode and Dark mode + blur. Is this expected? I believe this is not the case. Correct me if i'm wrong.

Thanks for spotting it. The problem was in the preview screen itself. It was using the same color as the loading skeleton in dark mode.
Light mode + blur is just white color with opacity, so should not be visible.

**Pull request merged**